network/src/protocols/discovery/mod.rs

```diff
@@ -274,25 +275,24 @@ where
     }
 
     // Creates DiscoveryMsg to be sent to some remote peer.
-    fn compose_discovery_msg(&self) -> DiscoveryMsg {
-        let mut msg = DiscoveryMsg::default();
-        for verified_note in self.known_peers.values() {
-            msg.notes.push(verified_note.raw_note.clone());
-        }
-        msg
+    fn compose_discovery_msg(&self) -> DiscoveryMsgProto {
+        let notes = self.known_peers.values().cloned().collect::<Vec<_>>();
+        let msg = DiscoveryMsg { notes };
+        let message = lcs::to_bytes(&msg).expect("LCS serialization failed");
```

Comment:
is serialization guaranteed to be infallible? should `lcs::to_bytes` reflect this?

Reply:
There were some other cases within networking that had expects that all likely ought to be handled. Bad practice on me for not fixing that :).

Reply:
This `.expect()` is acceptable IMO, since we would have verified the sanctity of everything stored in `known_peers` before storing it there.

Reply:
My preference would be that we remove all expect from this code base. Could we punt on this for now?

Reply:
yeah, let's punt